src/relay/backend/vm/compiler.cc

```diff
@@ -122,6 +123,49 @@ std::tuple<ConstMap, ConstTensorShapeMap> LayoutConstantPool(const Module& modul
 
 void InstructionPrint(std::ostream& os, const Instruction& instr);
 
+// Represent a runtime object that's going to be matched by pattern match expressions
+struct MatchValue {
+  virtual ~MatchValue() {}
+};
+using MatchValuePtr = std::shared_ptr<MatchValue>;
+
+// A runtime object that resides in a register
+struct RegisterValue : MatchValue {
+  // The register num
+  RegName rergister_num;
+
+  explicit RegisterValue(RegName reg) : rergister_num(reg) {}
+
+  ~RegisterValue() {}
+};
+
+// The value is a field of another runtime object
+struct AccessField : MatchValue {
+  MatchValuePtr parent;
+  // Field index
+  size_t index;
+  // Runtime register num after compiling the access field path
+  RegName reg{9999999};
```

Comment:
why has such a default? can this be a constructor param instead?

Reply:
I see. Can you make it -1 as it is a better default?

Reply:
Good catch! Changed to -1.